soh/src/overlays/actors/ovl_En_GirlA/z_en_girla.c

```diff
@@ -850,6 +911,23 @@ void EnGirlA_ItemGive_BottledItem(GlobalContext* globalCtx, EnGirlA* this) {
     Rupees_ChangeBy(-this->basePrice);
 }
 
+// This is called when EnGirlA_CanBuy_Randomizer returns CANBUY_RESULT_SUCCESS
+// The giving of the item is handled here, and no fanfare is played
+void EnGirlA_ItemGive_Randomizer(GlobalContext* globalCtx, EnGirlA* this) {
+    Player* player = GET_PLAYER(globalCtx);
+    ShopItemIdentity shopItemIdentity = Randomizer_IdentifyShopItem(globalCtx->sceneNum, this->randoSlotIndex);
+    GetItemEntry getItemEntry = Randomizer_GetItemFromKnownCheckWithoutObtainabilityCheck(shopItemIdentity.randomizerCheck, shopItemIdentity.ogItemId);
+
+    if (getItemEntry.modIndex == MOD_NONE) {
+        Item_Give(globalCtx, getItemEntry.itemId);
```

Comment:
This will need `gSaveContext.hasBgs = true` if the bought item is the BGS since Item_Give() doesn't handle that part.